packages/v3-sdk/src/functions/swap.ts

```diff
@@ -0,0 +1,238 @@
+import { CurrencyAmount, Token } from '@pancakeswap/sdk'
+import invariant from 'tiny-invariant'
+import { TICK_SPACINGS } from '../constants'
+import { NEGATIVE_ONE, ONE, ZERO } from '../internalConstants'
+import { LiquidityMath, SwapMath, TickMath } from '../utils'
+import { PoolState } from './getPool'
+import { hasInvolvedToken } from './hasInvolvedToken'
+
+interface StepComputations {
+  sqrtPriceStartX96: bigint
+  tickNext: number
+  initialized: boolean
+  sqrtPriceNextX96: bigint
+  amountIn: bigint
+  amountOut: bigint
+  feeAmount: bigint
+}
+
+export const swap = async ({
```

Comment:
btw I'm thinking whether we need a synchronous version of this `swap` function. 

It's async here simply because it's trying to look for the next initialized tick from asynchronous tick data provider. However, it's not efficient in practice, and for off chain routing what we need is simply a calculation given ticks data

Reply:
open for discussion on this

Reply:
how about optimize the `nextInitializedTickWithinOneWord` inner the tickDataProvider, 

or add a function to get next 10/20 Tick at once, then pass to `swap`